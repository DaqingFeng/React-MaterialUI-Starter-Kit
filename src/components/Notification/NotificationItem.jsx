import React from "react";
import PropTypes from "prop-types";
import Snackbar from "../Snackbar/Snackbar";
import NotificationType from "./NotificationType";
import NotificationItemStyle from "./jss";
import Button from "@material-ui/core/Button";
import { withStyles } from "@material-ui/core/styles";
import * as helper from "./helper";

@withStyles(NotificationItemStyle)
export default class NotificationItem extends React.Component {
  constructor(props, context) {
    super(props, context);
    this._timeout = undefined;
  }

  setCallbackTimeout = props => {
    const { onClose, open, autoClose, displayIn } = props;

    if (!onClose || open !== true || autoClose !== true || displayIn <= 0)
      return;

    if (this._timeout) return;
    this._timeout = setTimeout(this.onClosing, displayIn);
  };

  componentDidMount() {
    this.setCallbackTimeout(this.props);
  }

  componentWillReceiveProps(nextProps) {
    this.setCallbackTimeout(nextProps);
  }

  //Handling the internail closing event and call onClose with id
  onClosing = () => {
    const { onClose, id } = this.props;
    onClose({ id });
    this._timeout = undefined;
  };

  render() {
    const {
      id,
      type,
      message,
      icon,
      classes,
      displayIn,
      autoClose,
      onClick,
      ...others
    } = this.props;

    const color = helper.getColor(type);

    let finalIcon = icon;
    if (icon === true) finalIcon = helper.getIcon(type);

    return (
      <Snackbar
        {...others}
        key={id}
        onClose={this.onClosing}
        color={color}
        close={true}
        message={
          <Button onClick={onClick} className={classes.message}>
            {finalIcon}
            {" " + message}
          </Button>
        }
      />
    );
  }
}

NotificationItem.defaultProps = {
  place: "tr", //Top right
  displayIn: 4000,
  autoClose: true,
  type: NotificationType.INFO,
  open: true,
  icon: true
};

NotificationItem.propTypes = {
  //The Tye of notification
  type: PropTypes.oneOf([
    NotificationType.CONFIRM,
    NotificationType.DANGER,
    NotificationType.INFO,
    NotificationType.SUCCESS,
    NotificationType.WARNING
  ]),
  //The number of second will be displayed.
  displayIn: PropTypes.number,
  //Enable timeout to call onClose after displayIn second automatically.
  autoClose: PropTypes.bool,
  //The message of notification
  message: PropTypes.string.isRequired,
  //Close handler.
  onClose: PropTypes.func,
  //click event handler.
  onClick: PropTypes.func,
  //The icon of notification. set to false to hide the default icon.
  icon: PropTypes.oneOfType([PropTypes.element, PropTypes.bool])
};
