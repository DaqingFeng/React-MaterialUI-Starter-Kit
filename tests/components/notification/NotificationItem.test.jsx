import React from "react";
import ReactTestUtils from "react-dom/test-utils";
import { mount, shallow } from "enzyme";

import SvgIcon from "@material-ui/core/SvgIcon";
import NotificationItem from "../../../src/components/Notification/NotificationItem";
import Type from "../../../src/components/Notification/NotificationType";

const render = (renderMethod, { callBack, ...rest } = {}) => {
  if (!callBack) callBack = () => true;

  const item = renderMethod(
    <NotificationItem
      {...rest}
      closeNotification={callBack}
      message="Test Notification Item"
    />
  );

  if (item.mount) item.mount();
  return item;
};

const mountComponent = props => render(mount, props);
const shallowComponent = props => render(shallow, props);

const promiseComponent = ({ displayIn, ...rest }) =>
  new Promise((resolve, reject) => {
    let called = false;

    const item = mountComponent({
      ...rest,
      displayIn: displayIn,
      callBack: () => {
        called = true;
        resolve(called);
      }
    });

    setTimeout(() => {
      if (!called) reject(called);
    }, displayIn + 10);
  });

describe(`Testing ${NotificationItem.displayName} component`, () => {
  test("closeNotification will be called", () => {
    const pm = promiseComponent({ displayIn: 200 });
    return expect(pm).resolves.toBe(true);
  });

  test("if displayIn is 0 the closeNotification wont be called", () => {
    const pm = promiseComponent({ displayIn: 0 });
    return expect(pm).rejects.toBe(false);
  });

  test("update displayIn from 0 to 5 the closeNotification will be called", () => {
    // const item = mountComponent({ displayIn: 0 });
    // expect(item.instance().timeout).toBe(undefined);
    // item.setProps({ displayIn: 5 });
    // expect(item.instance().timeout).toBeDefined();
  });

  test(`render ${Type.INFO}`, () => {
    const item = mountComponent({ type: Type.INFO });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.CONFIRM}`, () => {
    const item = mountComponent({ type: Type.CONFIRM });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.DANGER}`, () => {
    const item = mountComponent({ type: Type.DANGER });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.SUCCESS}`, () => {
    const item = mountComponent({ type: Type.SUCCESS });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.WARNING}`, () => {
    const item = mountComponent({ type: Type.WARNING });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.WARNING} with custom Icon`, () => {
    const icon = (
      <SvgIcon>
        <path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z" />
      </SvgIcon>
    );

    const item = mountComponent({ type: Type.WARNING, icon });
    expect(item).toMatchSnapshot();
  });

  test(`render ${Type.WARNING} with Icon is undefined`, () => {
    const item = mountComponent({ type: Type.WARNING, icon: undefined });
    expect(item).toMatchSnapshot();
  });
});
