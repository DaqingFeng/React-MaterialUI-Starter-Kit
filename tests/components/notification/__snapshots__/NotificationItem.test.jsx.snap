// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing WithStyles(NotificationItem) component onClick test 1`] = `
<div
  className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-info-6"
    direction="down"
    role="alertdialog"
    style={Object {}}
  >
    <div
      className="MuiSnackbarContent-message-31 Snackbar-message-12"
    >
      <div>
        <span
          className=""
        >
          <span
            className="NotificationItem-message-2"
            onClick={null}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-84"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
             Test Notification Item
          </span>
        </span>
      </div>
    </div>
    <div
      className="MuiSnackbarContent-action-32"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root-97 MuiIconButton-root-91 MuiIconButton-colorInherit-92 Snackbar-iconButton-14"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-84 Snackbar-close-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-100"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing WithStyles(NotificationItem) component render confirm 1`] = `
<div
  className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-primary-10"
    direction="down"
    role="alertdialog"
    style={Object {}}
  >
    <div
      className="MuiSnackbarContent-message-31 Snackbar-message-12"
    >
      <div>
        <span
          className=""
        >
          <span
            className="NotificationItem-message-2"
            onClick={null}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-84"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
              />
            </svg>
             Test Notification Item
          </span>
        </span>
      </div>
    </div>
    <div
      className="MuiSnackbarContent-action-32"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root-97 MuiIconButton-root-91 MuiIconButton-colorInherit-92 Snackbar-iconButton-14"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-84 Snackbar-close-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-100"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing WithStyles(NotificationItem) component render danger 1`] = `
<div
  className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-danger-9"
    direction="down"
    role="alertdialog"
    style={Object {}}
  >
    <div
      className="MuiSnackbarContent-message-31 Snackbar-message-12"
    >
      <div>
        <span
          className=""
        >
          <span
            className="NotificationItem-message-2"
            onClick={null}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-84"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
              />
            </svg>
             Test Notification Item
          </span>
        </span>
      </div>
    </div>
    <div
      className="MuiSnackbarContent-action-32"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root-97 MuiIconButton-root-91 MuiIconButton-colorInherit-92 Snackbar-iconButton-14"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-84 Snackbar-close-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-100"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing WithStyles(NotificationItem) component render info 1`] = `
<div
  className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-info-6"
    direction="down"
    role="alertdialog"
    style={Object {}}
  >
    <div
      className="MuiSnackbarContent-message-31 Snackbar-message-12"
    >
      <div>
        <span
          className=""
        >
          <span
            className="NotificationItem-message-2"
            onClick={null}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-84"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
             Test Notification Item
          </span>
        </span>
      </div>
    </div>
    <div
      className="MuiSnackbarContent-action-32"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root-97 MuiIconButton-root-91 MuiIconButton-colorInherit-92 Snackbar-iconButton-14"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-84 Snackbar-close-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-100"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing WithStyles(NotificationItem) component render success 1`] = `
<div
  className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-success-7"
    direction="down"
    role="alertdialog"
    style={Object {}}
  >
    <div
      className="MuiSnackbarContent-message-31 Snackbar-message-12"
    >
      <div>
        <span
          className=""
        >
          <span
            className="NotificationItem-message-2"
            onClick={null}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-84"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
              />
            </svg>
             Test Notification Item
          </span>
        </span>
      </div>
    </div>
    <div
      className="MuiSnackbarContent-action-32"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root-97 MuiIconButton-root-91 MuiIconButton-colorInherit-92 Snackbar-iconButton-14"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-84 Snackbar-close-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-100"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing WithStyles(NotificationItem) component render warning 1`] = `
<div
  className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-warning-8"
    direction="down"
    role="alertdialog"
    style={Object {}}
  >
    <div
      className="MuiSnackbarContent-message-31 Snackbar-message-12"
    >
      <div>
        <span
          className=""
        >
          <span
            className="NotificationItem-message-2"
            onClick={null}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-84"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
              />
            </svg>
             Test Notification Item
          </span>
        </span>
      </div>
    </div>
    <div
      className="MuiSnackbarContent-action-32"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root-97 MuiIconButton-root-91 MuiIconButton-colorInherit-92 Snackbar-iconButton-14"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-84 Snackbar-close-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-100"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing WithStyles(NotificationItem) component render warning with Icon is undefined 1`] = `
<div
  className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-warning-8"
    direction="down"
    role="alertdialog"
    style={Object {}}
  >
    <div
      className="MuiSnackbarContent-message-31 Snackbar-message-12"
    >
      <div>
        <span
          className=""
        >
          <span
            className="NotificationItem-message-2"
            onClick={null}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-84"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
              />
            </svg>
             Test Notification Item
          </span>
        </span>
      </div>
    </div>
    <div
      className="MuiSnackbarContent-action-32"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root-97 MuiIconButton-root-91 MuiIconButton-colorInherit-92 Snackbar-iconButton-14"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-84 Snackbar-close-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-100"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing WithStyles(NotificationItem) component render warning with custom Icon 1`] = `
<div
  className="MuiSnackbar-root-23 MuiSnackbar-anchorOriginTopRight-26"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="MuiTypography-root-60 MuiTypography-body1-69 MuiPaper-root-33 MuiPaper-elevation6-41 MuiSnackbarContent-root-30 Snackbar-root-3 Snackbar-warning-8"
    direction="down"
    role="alertdialog"
    style={Object {}}
  >
    <div
      className="MuiSnackbarContent-message-31 Snackbar-message-12"
    >
      <div>
        <span
          className=""
        >
          <span
            className="NotificationItem-message-2"
            onClick={null}
          >
            <div
              id="icon"
            >
              Duy
            </div>
             Test Notification Item
          </span>
        </span>
      </div>
    </div>
    <div
      className="MuiSnackbarContent-action-32"
    >
      <button
        aria-label="Close"
        className="MuiButtonBase-root-97 MuiIconButton-root-91 MuiIconButton-colorInherit-92 Snackbar-iconButton-14"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-96"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-84 Snackbar-close-13"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <g>
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </g>
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-100"
        />
      </button>
    </div>
  </div>
</div>
`;
